ui/src/core_components/content/CoreLink.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+	<div class="CoreLink">
+		<a
+			:href="fields.url.value"
+			:target="fields.target.value"
+			:rel="fields.rel.value"
+		>
+			{{ displayText }}
+		</a>
+	</div>
+</template>
+
+<script lang="ts">
+import { FieldType } from "../../streamsyncTypes";
+import { cssClasses, primaryTextColor } from "../../renderer/sharedStyleFields";
+import injectionKeys from "../../injectionKeys";
+let options = [];
+
+export default {
+	streamsync: {
+		name: "Link",
+		description: "A component to create a hyperlink.",
+		category: "Content",
+		fields: {
+			url: {
+				name: "URL",
+				type: FieldType.Text,
+				desc: "A valid URL.",
```

Comment:
I was quite confused by the list of pages at first, when I got to try this component. Took me a while to be like "Ah, of course, the pages are accessible via #".

I love the new dynamic options capability, but it makes you think the right options are the ones listed, when I expect most people to enter a "standard" URL e.g. "github.com". 

If we keep it:
- I'd make it clearer in the `desc` like, "Specify a URL or choose a page."
- I'd set the default to `https://streamsync.cloud`. Bit of self promotion but it'd make it clear right away that yes, you can include "normal" URLs and not just #fragments.

I'm also not sure how it'd behave when the list is empty, needs checking. I expect most apps to not have any keyed pages.

---

Also, something to keep in mind... The hash can contain parameters that will be lost when switching pages this way.

https://www.streamsync.cloud/page-routes.html#routes-with-parameters

For example if you have `/#detailPage/product_id=32&country=AR` and you switch via event handler or [backend action](https://www.streamsync.cloud/backend-initiated-actions.html#changing-the-active-page) it'll preserve the rest of the hash.

If someone types `#mypage` then they literally asked for the parameters to be removed, but if they select it from a list they might have the expectation that it behaves like other page-jumping functionality.

---

I'm more inclined towards not keeping the dynamic options here, but they'd certainly come in handy for _Reuse Component_.

Reply:
The logic that manages the hash is in `CoreRoot.vue` if you're interested.

Reply:
I see that we've been explicitly asked to link pages. So maybe let's try how it'd look like with the minor changes I suggested under "If we keep it".

`desc` should probably be "Specify a URL or choose a page. Keep in mind that you can only link to pages for which a key has been specified."

Keen to hear your thoughts. Apologies for the verbosity, you probably weren't expecting an essay after submitting a PR for an `a href`.

Reply:
I think it's just how these kinds of links work. Personally, it would be really confusing for me if the link appended parameters at the end of the URL. I expect that most people using this app will have at least a really basic knowledge about HTML, and <a href> is probably the first thing they learn.